azure/components/baseline/setup.ftl

```diff
@@ -300,6 +300,30 @@
                 ]
               /]
 
+              [#-- Set static website hosting if an SPA is configured --]
+              [#local enableSiteHosting = false]
+              [#list blueprintObject.Tiers?values as tier ]
+                [#if (tier.Components!{})?has_content]
+                  [#list tier.Components as id,component]
+                    [#if component?keys?seq_contains(SPA_COMPONENT_TYPE)]
+                      [#-- found configured SPA, enable website hosting --]
+                      [#local enableSiteHosting = true]
+                    [/#if]
+                  [/#list]
+                [/#if]
+              [/#list]
```

Comment:
This feels magic, might be better as an attribute on the baseline with a check done by the SPA to see if its enabled.